### PR TITLE
Display registered tournaments in Contest Screen

### PR DESCRIPTION
**Pull request by Fouad-sys**
This Pull request adds the new Tournament interface and a corresponding TournamentAdapter. It also displays the registered tournaments on the Contest screen by fetching them from the database via the getTournaments function.

### CODE REVIEW

File: mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt [hunk 1]
```diff
@@ -1,79 +1,90 @@
...
+    currentUser: AuthenticatedUser,
     modifier: Modifier = Modifier,
     contentPadding: PaddingValues = PaddingValues(),
 ) {
-  ContestScreen(TournamentScreenState(), modifier, key = { it.uid }, contentPadding)
+  val tournamentFacade = LocalTournamentFacade.current
+  val scope = rememberCoroutineScope()
+
+  ContestScreen(
+      TournamentScreenState(currentUser, tournamentFacade, scope),
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt:86]: Missing a `remember(currentUser, tournamentFacade, scope) { ... }` ^^ ?

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/tournaments/TournamentFacade.kt [hunk 2]
```diff
@@ -11,6 +16,20 @@ import ch.epfl.sdp.mobile.infrastructure.persistence.store.Store
  * @param store the [Store] which is used to manage documents.
  */
 class TournamentFacade(private val auth: Auth, private val store: Store) {
+
+  /**
+   * Returns all of the registered tournaments of the application.
+   *
+   * @param user the current [AuthenticatedUser].
+   */
+  fun getTournaments(user: AuthenticatedUser): Flow<List<Tournament>> {
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/tournaments/TournamentFacade.kt:25]: I'd name this `tournaments(...)` to stay consistent with standard Kotlin naming.

Comment (Fouad-sys) [reply mobile/src/main/java/ch/epfl/sdp/mobile/application/tournaments/TournamentFacade.kt]: Fixed in 5d32a9fb42579af9f705b9889170f812091800b2

File: mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt (hunk 1 above)
Comment (Fouad-sys) [reply mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt]: Yes! Always forget! ^^
Fixed in 5d32a9fb42579af9f705b9889170f812091800b2

**Review by alexandrepiveteau — APPROVED** [review]

**Review by matt989253 — APPROVED** [review]

**Pull request merged**